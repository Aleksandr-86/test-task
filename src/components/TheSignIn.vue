<script setup lang="ts">
import BaseInput from '@/components/ui/BaseInput.vue'
import BaseCheckbox from '@/components/ui/BaseCheckbox.vue'
import { ref } from '@vue/reactivity'

const checkboxIsChecked = ref(false)
</script>

<template>
  <div>
    <div class="h-full">
      <!-- Left column container with background-->
      <div
        class="g-6 flex h-full flex-wrap items-center justify-center lg:justify-between">
        <!-- <div
          class="shrink-1 mb-12 grow-0 basis-auto md:mb-0 md:w-9/12 md:shrink-0 lg:w-6/12 xl:w-6/12">
          <img
            src="https://tecdn.b-cdn.net/img/Photos/new-templates/bootstrap-login-form/draw2.webp"
            class="w-full"
            alt="Sample image" />
        </div> -->

        <!-- Right column container -->
        <!-- <div class="mb-12 md:mb-0 md:w-8/12 lg:w-5/12 xl:w-5/12"> -->

        <div class="select-none">
          <div class="flex justify-center items-center space-x-3">
            <img class="h-12" src="icons/cloud-icon.png" />
            <p class="text-3xl">Aircloud</p>
          </div>

          <form class="w-[400px]">
            <div class="flex flex-col">
              <BaseInput
                label="Эл. почта"
                type="email"
                id="email"
                autocomplete="email"
                icon-src="icons/envelope.svg" />

              <BaseInput
                label="Пароль"
                type="password"
                id="password"
                autocomplete="current-password"
                icon-src="icons/key.svg"
                :show-password="true" />
            </div>

            <div class="mt-2">
              <BaseCheckbox
                label="Запомнить меня"
                v-model="checkboxIsChecked"
                :value="true" />

              <div class="flex items-center space-x-2 pl-1 bg-amber-200">
                <input class="w-4 h-4 form-checkbox rounded opacity-0" />
                <span class="text-lg">
                  Запомнить мои регистрационные данные для следующего раза.
                </span>
              </div>
            </div>

            <div class="mb-6 flex items-center justify-between">
              <!--Forgot password link-->
              <a href="#!">Forgot password?</a>
            </div>

            <!-- Login button -->
            <div class="text-center lg:text-left">
              <button
                type="button"
                class="inline-block rounded bg-primary px-7 pb-2.5 pt-3 text-sm font-medium uppercase leading-normal text-white shadow-[0_4px_9px_-4px_#3b71ca] transition duration-150 ease-in-out hover:bg-primary-600 hover:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] focus:bg-primary-600 focus:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)] focus:outline-none focus:ring-0 active:bg-primary-700 active:shadow-[0_8px_9px_-4px_rgba(59,113,202,0.3),0_4px_18px_0_rgba(59,113,202,0.2)]"
                data-te-ripple-init
                data-te-ripple-color="light">
                Login
              </button>

              <!-- Register link -->
              <p class="mb-0 mt-2 pt-1 text-sm font-semibold">
                Don't have an account?
                <a
                  href="#!"
                  class="text-danger transition duration-150 ease-in-out hover:text-danger-600 focus:text-danger-600 active:text-danger-700"
                  >Register</a
                >
              </p>
            </div>
          </form>
        </div>
      </div>
    </div>
  </div>
</template>

<style scoped></style>
