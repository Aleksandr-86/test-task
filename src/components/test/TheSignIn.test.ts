import { render } from '@testing-library/vue'
import TheSignIn from '@/components/TheSignIn.vue'

describe('Компонент TheSignIn', () => {
  it('структура соответствует снимку', () => {
    const wrapper = render(TheSignIn, {})

    expect(wrapper).toMatchSnapshot()
  })
})
