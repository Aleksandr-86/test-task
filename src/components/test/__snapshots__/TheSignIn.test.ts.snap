// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Компонент TheSignIn > структура соответствует снимку 1`] = `
{
  "baseElement": <body>
    <div>
      <div
        class="flex w-[353px] select-none flex-col items-center"
      >
        <div
          class="mb-2 mt-7 flex items-center justify-center space-x-3"
        >
          <div
            class="h-12 w-20"
          >
            <img
              src="/icons/cloud.png"
            />
          </div>
          <p
            class="text-3xl"
          >
            Облако
          </p>
        </div>
        <form
          class="bg-red flex flex-col"
        >
          <div
            class="flex flex-col gap-1"
          >
            <div
              class="flex flex-col space-y-2"
            >
              <div
                class="h-6"
              >
                <transition-stub
                  appear="false"
                  css="true"
                  enteractiveclass="transition linear duration-700"
                  enterfromclass="opacity-0"
                  leaveactiveclass="transition linear duration-700"
                  leavetoclass="opacity-0"
                  persisted="false"
                >
                  <!--v-if-->
                </transition-stub>
              </div>
              <div
                class="relative"
              >
                <input
                  autocomplete="email"
                  class="outline-solid h-12 w-full rounded text-3xl text-gray-800 shadow-[inset_2px_2px_4px_rgba(0,0,0,0.5)] outline-blue-800 dark:bg-gray-200 pe-3 ps-12"
                  id="email"
                  name="email"
                  placeholder="Эл. почта"
                  type="email"
                />
                <!-- Иконка слева -->
                <div
                  class="absolute bottom-0 left-0 top-0 flex p-2"
                >
                  <img
                    src="/icons/envelope.svg"
                  />
                </div>
                <!-- Иконка справа -->
                <transition-stub
                  appear="false"
                  css="true"
                  enteractiveclass="transition linear duration-700"
                  enterfromclass="opacity-0"
                  leaveactiveclass="transition linear duration-700"
                  leavetoclass="opacity-0"
                  persisted="false"
                >
                  <!--v-if-->
                </transition-stub>
              </div>
            </div>
            <div
              class="flex flex-col space-y-2"
            >
              <div
                class="h-6"
              >
                <transition-stub
                  appear="false"
                  css="true"
                  enteractiveclass="transition linear duration-700"
                  enterfromclass="opacity-0"
                  leaveactiveclass="transition linear duration-700"
                  leavetoclass="opacity-0"
                  persisted="false"
                >
                  <!--v-if-->
                </transition-stub>
              </div>
              <div
                class="relative"
              >
                <input
                  autocomplete="current-password"
                  class="outline-solid h-12 w-full rounded text-3xl text-gray-800 shadow-[inset_2px_2px_4px_rgba(0,0,0,0.5)] outline-blue-800 dark:bg-gray-200 px-12"
                  id="password"
                  name="password"
                  placeholder="Пароль"
                  type="password"
                />
                <!-- Иконка слева -->
                <div
                  class="absolute bottom-0 left-0 top-0 flex p-2"
                >
                  <img
                    src="/icons/key.svg"
                  />
                </div>
                <!-- Иконка справа -->
                <transition-stub
                  appear="false"
                  css="true"
                  enteractiveclass="transition linear duration-700"
                  enterfromclass="opacity-0"
                  leaveactiveclass="transition linear duration-700"
                  leavetoclass="opacity-0"
                  persisted="false"
                >
                  <!--v-if-->
                </transition-stub>
              </div>
            </div>
          </div>
          <div
            class="mt-5"
          >
            <div
              class="flex items-center space-x-2 px-1"
            >
              <input
                class="form-checkbox h-5 w-5 cursor-pointer rounded outline-blue-700 checked:outline-blue-700 focus:outline-blue-700"
                name="checkbox"
                type="checkbox"
                value="true"
              />
              <span
                class="font-bold"
              >
                Запомнить меня
              </span>
            </div>
            <div
              class="flex items-center pl-[32.5px]"
            >
              <span>
                 Запомнить мои регистрационные данные для следующего раза. 
              </span>
            </div>
          </div>
          <div
            class="flex justify-center"
          >
            <div
              class="mt-5 w-80"
            >
              <button
                class="flex h-12 w-full cursor-pointer items-center justify-center rounded-lg bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
                type="button"
              >
                <!--v-if-->
                 Войти
              </button>
            </div>
          </div>
          <div
            class="flex px-1 items-center before:mt-0.5 before:flex-1 before:border-t before:border-neutral-300 after:mt-0.5 after:flex-1 after:border-t after:border-neutral-300 mt-5"
          >
            <p
              class="mx-4 mb-0 text-center font-semibold"
            >
              Или
            </p>
          </div>
        </form>
        <div
          class="mb-5 mt-3 flex space-x-2"
        >
          <p
            class=""
          >
            Не имеете аккаунт?
          </p>
          <span>
            <routerlink
              class="font-bold text-blue-700 dark:text-blue-500"
              to="sign-up"
            >
              Зарегистрируйтесь!
            </routerlink>
          </span>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="flex w-[353px] select-none flex-col items-center"
    >
      <div
        class="mb-2 mt-7 flex items-center justify-center space-x-3"
      >
        <div
          class="h-12 w-20"
        >
          <img
            src="/icons/cloud.png"
          />
        </div>
        <p
          class="text-3xl"
        >
          Облако
        </p>
      </div>
      <form
        class="bg-red flex flex-col"
      >
        <div
          class="flex flex-col gap-1"
        >
          <div
            class="flex flex-col space-y-2"
          >
            <div
              class="h-6"
            >
              <transition-stub
                appear="false"
                css="true"
                enteractiveclass="transition linear duration-700"
                enterfromclass="opacity-0"
                leaveactiveclass="transition linear duration-700"
                leavetoclass="opacity-0"
                persisted="false"
              >
                <!--v-if-->
              </transition-stub>
            </div>
            <div
              class="relative"
            >
              <input
                autocomplete="email"
                class="outline-solid h-12 w-full rounded text-3xl text-gray-800 shadow-[inset_2px_2px_4px_rgba(0,0,0,0.5)] outline-blue-800 dark:bg-gray-200 pe-3 ps-12"
                id="email"
                name="email"
                placeholder="Эл. почта"
                type="email"
              />
              <!-- Иконка слева -->
              <div
                class="absolute bottom-0 left-0 top-0 flex p-2"
              >
                <img
                  src="/icons/envelope.svg"
                />
              </div>
              <!-- Иконка справа -->
              <transition-stub
                appear="false"
                css="true"
                enteractiveclass="transition linear duration-700"
                enterfromclass="opacity-0"
                leaveactiveclass="transition linear duration-700"
                leavetoclass="opacity-0"
                persisted="false"
              >
                <!--v-if-->
              </transition-stub>
            </div>
          </div>
          <div
            class="flex flex-col space-y-2"
          >
            <div
              class="h-6"
            >
              <transition-stub
                appear="false"
                css="true"
                enteractiveclass="transition linear duration-700"
                enterfromclass="opacity-0"
                leaveactiveclass="transition linear duration-700"
                leavetoclass="opacity-0"
                persisted="false"
              >
                <!--v-if-->
              </transition-stub>
            </div>
            <div
              class="relative"
            >
              <input
                autocomplete="current-password"
                class="outline-solid h-12 w-full rounded text-3xl text-gray-800 shadow-[inset_2px_2px_4px_rgba(0,0,0,0.5)] outline-blue-800 dark:bg-gray-200 px-12"
                id="password"
                name="password"
                placeholder="Пароль"
                type="password"
              />
              <!-- Иконка слева -->
              <div
                class="absolute bottom-0 left-0 top-0 flex p-2"
              >
                <img
                  src="/icons/key.svg"
                />
              </div>
              <!-- Иконка справа -->
              <transition-stub
                appear="false"
                css="true"
                enteractiveclass="transition linear duration-700"
                enterfromclass="opacity-0"
                leaveactiveclass="transition linear duration-700"
                leavetoclass="opacity-0"
                persisted="false"
              >
                <!--v-if-->
              </transition-stub>
            </div>
          </div>
        </div>
        <div
          class="mt-5"
        >
          <div
            class="flex items-center space-x-2 px-1"
          >
            <input
              class="form-checkbox h-5 w-5 cursor-pointer rounded outline-blue-700 checked:outline-blue-700 focus:outline-blue-700"
              name="checkbox"
              type="checkbox"
              value="true"
            />
            <span
              class="font-bold"
            >
              Запомнить меня
            </span>
          </div>
          <div
            class="flex items-center pl-[32.5px]"
          >
            <span>
               Запомнить мои регистрационные данные для следующего раза. 
            </span>
          </div>
        </div>
        <div
          class="flex justify-center"
        >
          <div
            class="mt-5 w-80"
          >
            <button
              class="flex h-12 w-full cursor-pointer items-center justify-center rounded-lg bg-blue-700 text-2xl font-bold text-gray-200 dark:bg-blue-100 dark:text-gray-700"
              type="button"
            >
              <!--v-if-->
               Войти
            </button>
          </div>
        </div>
        <div
          class="flex px-1 items-center before:mt-0.5 before:flex-1 before:border-t before:border-neutral-300 after:mt-0.5 after:flex-1 after:border-t after:border-neutral-300 mt-5"
        >
          <p
            class="mx-4 mb-0 text-center font-semibold"
          >
            Или
          </p>
        </div>
      </form>
      <div
        class="mb-5 mt-3 flex space-x-2"
      >
        <p
          class=""
        >
          Не имеете аккаунт?
        </p>
        <span>
          <routerlink
            class="font-bold text-blue-700 dark:text-blue-500"
            to="sign-up"
          >
            Зарегистрируйтесь!
          </routerlink>
        </span>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "emitted": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "html": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
